src/link/link.c

```diff
@@ -148,7 +148,19 @@ size_t _z_link_recv_exact_zbuf(const _z_link_t *link, _z_zbuf_t *zbf, size_t len
 
 int8_t _z_link_send_wbuf(const _z_link_t *link, const _z_wbuf_t *wbf) {
     int8_t ret = _Z_RES_OK;
-    _Bool link_is_streamed = _Z_LINK_IS_STREAMED(link->_capabilities);
+    _Bool link_is_streamed = false;
+
+    switch (link->_capabilities) {
```

Comment:
This should be a bitwise operation as defined in the existing `_Z_LINK_IS_STREAMED`

Reply:
Future transports will support or not a given set of functionalities. And the way, the core code should be handled is accordingly to the support or not these functionalities.
As far as I remember, this is how the Rust implementation is also tackling this issue.

Can you give an example of a future transport that does not map? Note that if new properties must be added, they can be done is a clean way with the current approach.

Reply:
tagging @Mallets

Reply:
In accordance with my previous comment, bitwise operations have the tendance to be interpreted as a combination of indipendent parameters. This has led to some confusion and misconception on how things should be done and structured. I believe in this case having a clear variant separation helps in understanding what are the real cardinality of available options.

Reply:
These parameters are and should be interpreted independently.

If you notice on your own review, this new approach has in fact introduced confusion and misconceptions. @jean-roland was considering `Z_LINK_CAP_MULTICAST_STREAM` as a possible capability that, in your own words, does not make much sense.

Reply:
Let me know what is your decision, so I can remove or keep the requested changes in my previous code review.

Reply:
That's because bitfields are interpreted as combinatory parameters and the enum was defined from that combination of bits. As a result, the invalid variant was added.

Reply:
The usage here is intended to be a flag structure.

Moreover, if the scope of the refactor is to ease future additions, I find the switch case approach counter-intuitive since it will require to modify more parts that were supposed to be isolated for a given capability.

I would ask @p-avital opinion here as well.

Reply:
I added MULTICAST_STREAM only because bluetooth was declared multicast and streamed.

Reply:
Then I withdraw my comment: ``As far as I know there are no multicast protocols that are byte-streamed.`` 
Now I know :)

Reply:
My take on this is that regardless of the existence of a multicast-streamed transport, I think we always treat multicast-ness and streamed-ness independently (since one affects routing and the other serialization), so while the enum might highlight that some don't exist, is that really a relevant fact we need to encode in here?

In general, I take an enum going through the combinations of X and Y as `X_Y` to be a sign that these properties are rather independent, angling toward the bitfield. But bitfields are a bit ugly: I like the register-struct proposal @jean-roland made in the other thread

Reply:
Then I propose to move forward with @jean-roland proposal then.